code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/utility/StringEncoder.java

```diff
@@ -96,4 +121,107 @@ private static int getFnv1aHash(final String input) {
 
         return hash;
     }
+
+    /**
+     * Converts a string to hexadecimal notation
+     * @param originalString the string for which hexadecimal value is to be computed
+     * @return hexadecimal representation of the string if valid, null otherwise
+     */
+    public static String getHexString(final String originalString) {
+        if (StringUtils.isNullOrEmpty(originalString)) {
+            return null;
+        }
+
+        byte[] bytes;
+
+        try {
+            bytes = originalString.getBytes(CHARSET_UTF_8);
+        } catch (UnsupportedEncodingException ex) {
+            MobileCore.log(LoggingMode.WARNING, LOG_TAG, "Failed to get hex from string " + ex.getMessage());
+            return null;
+        }
+
+        final int bytesLength = bytes.length;
+        final char[] chars = new char[bytesLength << 1];
+        int hexIndex;
+        int index = 0;
+        int offset = 0;
+
+        while (offset < bytesLength) {
+            hexIndex = (bytes[offset++] & OXFF) << 1;
+            chars[index++] = BYTE_TO_HEX[hexIndex++];
+            chars[index++] = BYTE_TO_HEX[hexIndex];
+        }
+
+        return new String(chars);
+    }
+
+    /**
+     * Decodes a hexadecimal string
+     *
+     * @param hexString the hexadecimal string to be decoded
+     * @return decoded hexadecimal string if valid, null otherwise
+     */
+    public static String hexToString(final String hexString) {
+        if (hexString == null || hexString.length() <= 0 || hexString.length() % 2 != 0) {
+            return null;
+        }
+
+        final int length = hexString.length();
+        byte[] data = new byte[length / 2];
+
+        for (int i = 0; i < length; i += 2) {
+            final int radix = 16;
+            final int fourDigit = 4;
+            data[i / 2] = (byte) ((Character.digit(hexString.charAt(i), radix) << fourDigit) +
+                    Character.digit(hexString.charAt(i + 1), radix));
+        }
+
+        String decodedString = null;
+
+        try {
+            decodedString = new String(data, CHARSET_UTF_8);
```

Comment:
Can we switch to calling another String constructor (`public String (byte[] bytes, Charset charset)`) as below?
https://developer.android.com/reference/java/lang/String?hl=en#String(byte[],%20java.nio.charset.Charset)
Also, we can remove the public constant `CHARSET_UTF_8`.
```java
decodedString = new String(data, StandardCharsets.UTF_8);
```
> **StandardCharsets** was added since Android API level 19